stdlib/2/pydoc.pyi

```diff
@@ -0,0 +1,176 @@
+# Stubs for pydoc (Python 2)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from typing import Any, AnyStr, Callable, IO, List, Mapping, MutableMapping, Optional, Tuple, Type, Union
+from repr import Repr
+
+from types import FunctionType, MethodType, ModuleType, TracebackType
+# the return type of sys.exc_info(), used by ErrorDuringImport.__init__
+_Exc_Info = Tuple[Optional[Type[BaseException]], Optional[BaseException], Optional[TracebackType]]
+
+__author__ = ...  # type: str
+__date__ = ...  # type: str
+__version__ = ...  # type: str
+__credits__ = ...  # type: str
+
+def pathdirs() -> List[str]: ...
+def getdoc(object: object) -> str: ...
+def splitdoc(doc) -> Tuple[str, str]: ...
+def classname(object: object, modname: str) -> str: ...
+def isdata(object: object) -> bool: ...
+def replace(text: str, *pairs: List[str]) -> str: ...
+def cram(text: str, maxlen: int) -> str: ...
+def stripid(text: str) -> str: ...
+def allmethods(cl: type) -> MutableMapping[str, MethodType]: ...
+def visiblename(name: str, all: Optional[List[str]] = ..., obj: Optional[object] = ...) -> bool: ...
+def classify_class_attrs(object: object) -> List[Tuple[str, str, type, str]]: ...
+
+def ispackage(path: str) -> bool: ...
+def source_synopsis(file: IO[AnyStr]) -> Optional[str]: ...
+def synopsis(filename: str, cache: MutableMapping[str, Tuple[int, str]] = ...) -> str: ...
+
+class ErrorDuringImport(Exception):
+    filename = ...  # type: str
+    exc = ...  # type: Optional[Type[BaseException]]
+    value = ...  # type: Optional[BaseException]
+    tb = ...  # type: Optional[TracebackType]
+    def __init__(self, filename: str, exc_info: _Exc_Info) -> None: ...
+
+def importfile(path: str) -> ModuleType: ...
+def safeimport(path: str, forceload: bool = ..., cache: MutableMapping[str, ModuleType] = ...) -> ModuleType: ...
+
+class Doc:
+    def document(self, object: object, name: Optional[str] = ..., *args: Any) -> str: ...
+    def fail(self, object: object, name: Optional[str] = ..., *args: Any) -> str: ...
+    def docmodule(self, object: object, name: Optional[str] = ..., *args: Any) -> str: ...
```

Comment:
The code _should_ be written with Doc as an abstract base class for HTMLDoc and TextDoc. Instead, it's a concrete class that fails when you try to use it. (Because Python didn't have ABCs when this was written, right?) Making all of the `doc*` methods strict aliases of `fail` breaks the subclasses, which _do_ return actual values from their `doc*` methods.